lib/ansible/plugins/callback/grafana_annotations.py

```diff
@@ -0,0 +1,234 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import json
+import socket
+import getpass
+from base64 import b64encode
+from datetime import datetime
+
+from ansible.module_utils.urls import open_url
+from ansible.plugins.callback import CallbackBase
+
+DOCUMENTATION = '''
+    callback: grafana_annotations
+    type: notification
+    short_description: send ansible events as annotations on charts to grafana over http api.
+    description:
+      - This callback will report start, failed and stats events to Grafana as annotations (https://grafana.com)
+    version_added: "2.5"
+    requirements:
+      - whitelisting in configuration
+    options:
+      grafana_url:
+        description: Grafana annotations api URL
+        env:
+          - name: GRAFANA_URL
+        default: http://127.0.0.1:3000/api/annotations
+      grafana_validate_certs:
+        description: (bool) validate the SSL certificate of the Grafana server. (For HTTPS url)
+        env:
+          - name: GRAFANA_VALIDATE_CERT
+        default: True
+      http_agent:
+        description: The HTTP 'User-agent' value to set in HTTP requets.
+        env:
+          - name: HTTP_AGENT
+        default: 'Ansible (grafana_annotations callback)'
+      api_key:
+        description: Grafana API key, allowing to authenticate when posting on the HTTP API.
+                     If not provided, grafana_login and grafana_password will
+                     be required.
+        env:
+          - name: GRAFANA_API_KEY
+      grafana_user:
+        description: Grafana user used for authentication. Ignored if api_key is provided.
+        env:
+          - name: GRAFANA_USER
+      grafana_password:
+        description: Grafana password used for authentication. Ignored if api_key is provided.
+        env:
+          - name: GRAFANA_PASSWORD
+      grafana_dashboard_id:
+        description: The grafana dashboard id where the annotation shall be created.
+        env:
+          - name: GRAFANA_DASHBOARD_ID
+      grafana_panel_id:
+        description: The grafana panel id where the annotation shall be created.
+        env:
+          - name: GRAFANA_PANEL_ID
+'''
+
+
+PLAYBOOK_START_TXT = """\
+Started playbook {playbook}
+
+From '{hostname}'
+By user '{username}'
+"""
+
+PLAYBOOK_ERROR_TXT = """\
+Playbook {playbook} Failure !
+
+From '{hostname}'
+By user '{username}'
+
+'{task}' failed on {host}
+
+debug: {result}
+"""
+
+PLAYBOOK_STATS_TXT = """\
+Playbook {playbook}
+Duration: {duration}
+Status: {status}
+
+From '{hostname}'
+By user '{username}'
+
+Result:
+{summary}
+"""
+
+
+def to_millis(dt):
+    return int(dt.strftime('%s')) * 1000
+
+
+def str2bool(string):
```

Comment:
When you switch the callback to use the `get_option` API. This function can go away, as the ansible config system can already handle this.